samples/MvcSample.Web/wwwroot/Scripts/Mvc.js

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+var MVC = (function () {
+    // Takes the data which needs to be converted to MVC understandable format.
+    var _stringify = function (data) {
+        // This holds the stringified result.
+        var result = "";
+        for (var element in data) {
+            result += process(element, data[element]);
+        }
+
+        // An '&' is appended at the end. Removing it.
+        return result.slice(0, -1);
+    }
+
+    function process(key, value, prefix) {
+        // Ignore functions.
+        if (typeof value === "function") {
+            return;
+        }
+
+        if (Object.prototype.toString.call(value) === '[object Array]') {
+            var result = "";
+            for (var i = 0; i < value.length; i++) {
+                var tempPrefix = (prefix == null ? key : prefix) + "[" + i + "]";
```

Comment:
You could do `prefix || key`. Alternatively, you might have to do a falsey value check `tempPrefix = !prefix ? key : prefix` to account for `undefined`. The first one's easier to read.